### PR TITLE
A bunch of changes. Most notable is a much improved Accordion with tests!

### PR DESCRIPTION
**Pull request by n8mellis**
I also fixed a bunch of linting errors, updated the versions of several libraries, and standardized some of the documentation.

Also of particular note, I updated the `BaseContainer` to automatically render the Example, Code, and Documentation sections separately.  This changes means that the class is now designed to be subclassed rather than used via composition.  The `AccordionContainer` class has been updated to show an example.  I would crave your feedback on this approach.  For more information on my thinking, look at the individual commit message that change was made in.

Apologies in advance for the jumbled nature of this pull request.  I meant to split all these out into separate pull requests, but they're so interdependent now that it was more trouble than I wanted to go into.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-var gulp = require('gulp'),
+let gulp = require('gulp'),
```
Comment (mstanaland) [inline gulpfile.js:1]: This works without renaming to `gulpfile.babel.js`? And if we're going to use ES2015, then why not import?

**Review by mstanaland — COMMENTED** [review]
The like the changes to the BaseContainer ... keeps things more organized. But I don't think it's flexible enough. For example, Form controls has multiple display areas, code samples, and descriptions. It's like we need the ability to have multiple baseContainers on one page. 

I'm also uncertain the page format standards.usa.gov is using (and we're currently replicating) works.  We're building code documentation for developers, so initially hiding code and separating it from the display and documentation seems not right. 18F's standards.usa.gov is documenting style and style philosophies, so the code being more secondary is fine. 

I also think it doesn't make sense to replicate 18F's content in their 'Documentation' tabs. But we do need to document each component: how to import it, what props are required, what props are optional, what features it supports and basic usage. Possibly also what accessibility methods are being used behind the scenes (i.e. something like "adds aria-disabled' if disabled={true}".

I think we need a format that makes it really easy to get all that info, not take up tons of space, and doesn't involve clicking accordions. And it needs to be flexible enough for groups of inter-dependent components. Take Tables as an example ... Tables should be a single page, but it's really several components (Table, Row, TableCell, TableHeaderCell, etc), each needing documentation.

Does that make sense? Do you disagree? Also I'm fine if we punt on this ... use the format in this pull request for now and adjust later.

File: gulpfile.js (hunk 1 above)
Comment (n8mellis) [reply gulpfile.js:1]: Excellent questions.  I believe it still works but I'll test it.  As for import, I've never really seen import being used in gulpfiles before.  I prefer the import syntax over the require syntax so if we decide that's how we want to do it, then you'll hear no objections from me.

**Comment by n8mellis** [comment]
I agree with everything you said.  Great observations.  To date, I haven't tried to do too much re-designing of the styleguide but you make great points here; especially about including code documentation.

Clearly you are right that this BaseContainer refactor is not flexible enough.  You are also clearly right that we'll need to support multiple copies in individual pages.  That makes the argument in favor of going back to composition and maybe creating a SampleCode component, etc. for those individual pieces and then we can mix and match as desired.  I'm open to other suggestions as well.

In the not-too-distant-future, I'd like to support auto-generating JSDocs from the documentation that we've been writing.  That should be the primary point of reference for developers.  Whether we blend the styleguide and the developer guide together remains an open question.

What do others think?

**Comment by vadnguyen** [comment]
I agree with all of that @mstanaland. We need a new format that does not involve an additional accordion to view the component documentation and we need content update for the documentation.

**Comment by n8mellis** [comment]
I'm going to merge this now, but we will replace my new BaseContainer with @mstanaland's in a future pull request.

**Pull request merged**